tsa-core/src/main/kotlin/org/usvm/test/resolver/TvmTestStateResolver.kt

```diff
@@ -512,9 +513,15 @@ class TvmTestStateResolver(
             val keySet = mutableSetOf<UExpr<UBvSort>>()
             val resultEntries = mutableMapOf<TvmTestIntegerValue, TvmTestSliceValue>()
 
+            if (dict is UConcreteHeapRef) {
+                val inputDict = state.inputDictionaryStorage.memory[dict]
+                if (inputDict != null) {
+                    return handleInputDict(inputDict, dict, dictId, keyLength, modelRef)
+                }
+            }
```

Comment:
And what if `dict` is ITE where one of the leaves is input dict? It can be input dict in the model. (You should probably first eval `dict`)

Reply:
Fixed it by using only concrete ref